From 4c96ac61821e798f8ec5ecf3f4b4a54123f84876 Mon Sep 17 00:00:00 2001
From: Andreas Doehler <andreas.doehler@gmail.com>
Date: Wed, 18 Mar 2015 14:47:38 +0100
Subject: [PATCH] Custom Host Notes are broken for the "all_services" view.
 Made extra function for custom_host_notes and custom_svc_notes.

---
 web/plugins/views/painters.py | 47 +++++++++++++++++++++++++++++++++----------
 1 file changed, 36 insertions(+), 11 deletions(-)

diff --git a/web/plugins/views/painters.py b/web/plugins/views/painters.py
index 6b66dde..f70a9df 100644
--- a/web/plugins/views/painters.py
+++ b/web/plugins/views/painters.py
@@ -887,16 +887,41 @@ def notes_matching_pattern_entries(dirs, item):
                     matching.append(dir + "/" + pattern)
     return matching
 
-def paint_custom_notes(row):
+def paint_custom_svc_notes(row):
     host = row["host_name"]
     svc = row.get("service_description")
-    if svc:
-        notes_dir = defaults.default_config_dir + "/notes/services"
-        dirs = notes_matching_pattern_entries([notes_dir], host)
-        item = svc
-    else:
-        dirs = [ defaults.default_config_dir + "/notes/hosts" ]
-        item = host
+    notes_dir = defaults.default_config_dir + "/notes/services"
+    dirs = notes_matching_pattern_entries([notes_dir], host)
+    item = svc
+
+    files = notes_matching_pattern_entries(dirs, item)
+    files.sort()
+    files.reverse()
+    contents = []
+    def replace_tags(text):
+        sitename = row["site"]
+        site = html.site_status[sitename]["site"]
+        return text\
+            .replace('$URL_PREFIX$',     site["url_prefix"])\
+            .replace('$SITE$',           sitename)\
+            .replace('$HOSTNAME$',       host)\
+            .replace('$HOSTNAME_LOWER$', host.lower())\
+            .replace('$HOSTNAME_UPPER$', host.upper())\
+            .replace('$HOSTNAME_TITLE$', host[0].upper() + host[1:].lower())\
+            .replace('$HOSTADDRESS$',    row["host_address"])\
+            .replace('$SERVICEOUTPUT$',  row.get("service_plugin_output", ""))\
+            .replace('$HOSTOUTPUT$',     row.get("host_plugin_output", ""))\
+            .replace('$SERVICEDESC$',    row.get("service_description", ""))
+
+    for f in files:
+        contents.append(replace_tags(unicode(file(f).read(), "utf-8").strip()))
+    return "", "<hr>".join(contents)
+
+def paint_custom_host_notes(row):
+    host = row["host_name"]
+    svc = row.get("service_description")
+    dirs = [ defaults.default_config_dir + "/notes/hosts" ]
+    item = host
 
     files = notes_matching_pattern_entries(dirs, item)
     files.sort()
@@ -909,7 +934,7 @@ def paint_custom_notes(row):
             .replace('$URL_PREFIX$',     site["url_prefix"])\
             .replace('$SITE$',           sitename)\
             .replace('$HOSTNAME$',       host)\
-	    .replace('$HOSTNAME_LOWER$', host.lower())\
+            .replace('$HOSTNAME_LOWER$', host.lower())\
             .replace('$HOSTNAME_UPPER$', host.upper())\
             .replace('$HOSTNAME_TITLE$', host[0].upper() + host[1:].lower())\
             .replace('$HOSTADDRESS$',    row["host_address"])\
@@ -925,7 +950,7 @@ multisite_painters["svc_custom_notes"] = {
     "title"   : _("Custom services notes"),
     "short"   : _("Notes"),
     "columns" : [ "host_name", "host_address", "service_description", "service_plugin_output" ],
-    "paint"   : paint_custom_notes,
+    "paint"   : paint_custom_svc_notes,
 }
 
 multisite_painters["svc_staleness"] = {
@@ -1391,7 +1416,7 @@ multisite_painters["host_custom_notes"] = {
     "title"   : _("Custom host notes"),
     "short"   : _("Notes"),
     "columns" : [ "host_name", "host_address", "host_plugin_output" ],
-    "paint"   : paint_custom_notes,
+    "paint"   : paint_custom_host_notes,
 }
 
 multisite_painters["host_comments"] = {
-- 
1.9.5.msysgit.0

